MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -1,231 +1,152 @@
+#import <sqlite3.h>
+
 #import "MSDBStoragePrivate.h"
-#import "MSDatabaseConnection.h"
-#import "MSLogger.h"
 #import "MSMobileCenterInternal.h"
-#import "MSSqliteConnection.h"
-#import "MSUtility.h"
+#import "MSStorage.h"
+#import "MSUtility+File.h"
 
 @implementation MSDBStorage
 
-#pragma mark - Initialization
+- (instancetype)initWithSchema:(MSDBSchema *)schema filename:(NSString *)filename {
+  if ((self = [super init])) {
+    NSMutableArray *tableQueries = [NSMutableArray new];
+    NSMutableDictionary *dbColumnsIndexes = [NSMutableDictionary new];
+
+    // Path to the database.
+    NSURL *appSupportURL = [[[NSFileManager defaultManager] URLsForDirectory:NSApplicationSupportDirectory
+                                                                   inDomains:NSUserDomainMask] lastObject];
+    if (appSupportURL) {
+      NSURL *fileURL =
+          [appSupportURL URLByAppendingPathComponent:[kMSStorageDirectory stringByAppendingPathComponent:filename]];
+      [MSUtility createFileAtURL:fileURL];
+      _filePath = fileURL.absoluteString;
+    }
 
-- (instancetype)init {
-  self = [super init];
-  if (self) {
-    _connection = [[MSSqliteConnection alloc] initWithDatabaseFilename:kMSDBFileName];
-    _batches = [NSMutableDictionary<NSString *, NSArray<NSString *> *> new];
-    _capacity = NSUIntegerMax;
+    // Browse tables.
+    for (NSString *tableName in schema) {
+      NSMutableDictionary *tableColumnsIndexes = [NSMutableDictionary new];
+      NSMutableArray *columnQueries = [NSMutableArray new];
+      NSArray<NSDictionary *> *columns = schema[tableName];
+
+      // Optimization, don't even compute the query if the table already exists.
+      BOOL tableDontExist = ![self tableExists:tableName];
+
+      // Browse columns.
+      for (NSUInteger i = 0; i < columns.count; i++) {
+        NSString *columnName = columns[i].allKeys[0];
+        [tableColumnsIndexes setObject:@(i) forKey:columnName];
+
+        // Compute column query.
+        if (tableDontExist) {
+          [columnQueries addObject:[NSString stringWithFormat:@"\"%@\" %@", columnName,
+                                                              [columns[i][columnName] componentsJoinedByString:@" "]]];
+        }
+      }
+
+      // Compute table query.
+      if (tableDontExist) {
+        [tableQueries addObject:[NSString stringWithFormat:@"CREATE TABLE \"%@\" (%@);", tableName,
```

Comment:
It's a good point, that can be an upgrade to this class, we don't need it as for now. So I suggest we leave it for now and tackle it later as needed.